### PR TITLE
fix: session token not uri conform at all times

### PR DESCRIPTION
**Pull request by BeierKevin**
# PR description

Added URI encoding to the session token, to handle errors or problems when copying or trying to use the token.

# Definition Of Done (DoD)

_This PR can be squashed / merged if_

-   [x] a developer is assigned
-   [x] the PR is NOT estimated
-   [x] the PR is labeled
-   [x] the PR is NOT assigned to the current sprint
-   [x] a meaningful title has been set according to https://www.conventionalcommits.org/
-   [x] the PR is described in detail
-   [x] the PR links to an issue
-   [ ] the PR has been reviewed

_Add additional conditions here if necessary for this PR_

fix: #190

### CODE REVIEW

File: src/components/SettingsPrompt/SettingsPrompt.vue [hunk 1]
```diff
@@ -58,6 +58,8 @@
                             variant="outlined"
                             label="Access Token"
                             name="apiAccessToken"
+                            hint="Token will be URI encoded on when unselected"
```
Comment (Claiyc) [inline src/components/SettingsPrompt/SettingsPrompt.vue:61]:
```suggestion
                            hint="Token will automatically be URI encoded"
```

**Review by Claiyc — CHANGES_REQUESTED** [review]
Does not seem to work (newly generated token contains not allowed characters):
image

Pasting & unselecting works fine, however here are a few takeaways that would make it more intuitive:

- URI encode right on paste (if necessary)
- URI encode on session start (if necessary) -> we need to DEFINITELY make sure it's URI conform, only encoding on user input is not failsafe

**Comment by BeierKevin** [comment]
There is a fallback in the Vigad.ts file which will encode it
image

It will get URI encoded on session start, because when you do anything outside of the input feel like interacting in this example starting the session via the button press the input will be encoded, but i can add another failsafe so if the start session button is pressed it checks if the input is uri conform and encode it if neccessarry. 

About the paste inside the input field i can have a look into it

**Comment by Claiyc** [comment]
> There is a fallback in the Vigad.ts file which will encode it image
> 
> It will get URI encoded on session start, because when you do anything outside of the input feel like interacting in this example starting the session via the button press the input will be encoded, but i can add another failsafe so if the start session button is pressed it checks if the input is uri conform and encode it if neccessarry.
> 
> About the paste inside the input field i can have a look into it

fine by me if it somehow gets encoded on start. However when encoding on start, the content inside of the input needs to be encoded as well (replaced with the encoded value)

**Comment by Claiyc** [comment]
Main issue is not fixed yet (generated token not encoded)

File: src/composables/useSession/useSession.ts [hunk 2]
```diff
@@ -27,10 +40,37 @@ export default function useSession() {
...
+     */
+    const generateSessionToken = (): void => {
+        sessionToken.value = generateValidToken();
+        encodeSessionToken();
+    };
+
+    /**
+     * Encodes the session token if it is not already encoded by default
+     */
+    const encodeSessionToken = (): void => {
```
Comment (Claiyc) [inline src/composables/useSession/useSession.ts:54]: '%' is actuall not valid itself (this function seems to not encode this correctly)

Comment (BeierKevin) [reply src/composables/useSession/useSession.ts:54]: image

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/decodeURIComponent

idk what the workaround could be, a single % does not really work, or atleast there is not an elegant solution

Comment (Claiyc) [reply src/composables/useSession/useSession.ts:54]: that snipped seems like a good direction. The thing is that % as itself is not URI conform, however %D1 for example is (that is the encoding for some special character)

Comment (BeierKevin) [reply src/composables/useSession/useSession.ts:54]: Yeah i think i just need to inform the user then that the provided input is not URI conform basically

**Review by Claiyc — APPROVED** [review]

**Pull request merged**